06-building-a-program-start-to-finish/writing-tests/output/InvertedIndexSpec.js

```diff
@@ -0,0 +1,345 @@
+  });
+
+  describe('The readFile method', () => {
+    it('should return the json file if it is valid', () => {
+      expect(InvertedIndex.readFile(JSON.stringify(books))).toEqual((books));
+    });
+
+    it('should return the json file for an empty json', () => {
+      expect(InvertedIndex.readFile(JSON.stringify(bookEmptyJson)))
+        .toEqual((bookEmptyJson));
+    });
+
+    it('should return false if the file cannot be parsed', () => {
+      expect(InvertedIndex.readFile((bookText))).toBe((false));
+      expect(InvertedIndex.readFile((bookText))).toBeFalsy();
+    });
+  });
+
+  describe('The validateFile method', () => {
+    describe('ensures each document is valid', () => {
+      let result = '';
+      beforeEach((done) => {
+        InvertedIndex.validateFile(bookBadJson, 'bookBadJson.json')
+        .catch((error) => {
+          result = error;
+          done();
+        });
+      });
+      it('should have a `title` and `text` properties', () => {
+        const expected = "No 'title' or 'text' in Document 1 of bookBadJson";
+        expect(result).toBe(expected);
+        expect(typeof result).toBe('string');
+      });
+    });
+
+    describe('title and text properties must have valid contents', () => {
+      let result = '';
+      beforeEach((done) => {
+        InvertedIndex.validateFile(booksInvalidContent, 'booksInvalidContent')
+        .catch((error) => {
+          result = error;
+          done();
+        });
+      });
+      it('both title and text must contain values', () => {
+        expect(result).toBe('Document 1 have an empty title or text.');
+        expect(typeof result).toBe('string');
+      });
+    });
+
+    describe('title and text properties must have valid contents', () => {
+      let result = '';
+      beforeEach((done) => {
+        InvertedIndex.validateFile(bookEmptyJson, 'bookEmptyJson.json')
+        .catch((error) => {
+          result = error;
+          done();
+        });
+      });
+      it('both title and text must contain values', () => {
+        expect(result).toBe('Cannot index an empty object');
+        expect(typeof result).toBe('string');
+      });
+    });
+
+    describe('title and text properties must have valid contents', () => {
+      let result = '';
+      beforeEach((done) => {
+        InvertedIndex.validateFile(books, 'books.json')
+        .then((response) => {
+          result = response;
+          done();
+        });
+      });
+      it('both title and text must contain values', () => {
+        expect(result).toEqual({ books: {
+          0: {
+            title: 'how to read a book',
+            text: 'men are rational animals. their rationality ' +
+                  'agreement is the source of their power to agree. ',
+          },
+          1: {
+            title: 'the naked ape.',
+            text: 'indeed, we have the most subtle and complex' +
+                  ' facial expression system of all living animals.',
+          },
```

Comment:
Please use template literals here instead of concatenating this string or put the entire object in a file and reference it.